compiler/luci/import/src/Nodes/CircleConst.cpp

```diff
@@ -67,6 +67,44 @@ void copy_data(const VectorWrapper<uint8_t> &raw_data, uint32_t num_elements,
   }
 }
 
+template <>
+void copy_data<loco::DataType::S4>(const VectorWrapper<uint8_t> &raw_data, uint32_t num_elements,
+                                   CircleConst *const_node)
+{
+  // TODO support sparse?
+  assert(const_node->sparsityparam() == nullptr);
+  if (const_node->sparsityparam())
+    return;
+
+  uint32_t raw_size = (num_elements + 1) / 2;
+  assert(raw_data.size() == raw_size);
+
+  const uint8_t *data = raw_data.data();
+
+  printf("=== raw data\r\n");
+  for (uint32_t i = 0; i < raw_size; ++i)
+  {
+    printf("%2d: 0x%02hhx\r\n", i, data[i]);
+  }
+
+  const_node->size<loco::DataType::S4>(num_elements);
+  for (uint32_t i = 0; i < raw_size; ++i)
+  {
+    uint32_t idx = i * 2;
+    // 1bit for sign, 3bit for value
+    const_node->at<loco::DataType::S4>(idx++) = static_cast<int8_t>(data[i] << 4) >> 4;
+    if (idx < num_elements)
+      const_node->at<loco::DataType::S4>(idx) = static_cast<int8_t>(data[i]) >> 4;
```

Comment:
## How packed int4(s4) values are converted to int16(s16)?

Let's assume two s4 values (AXXX, BYYY) are packed in S8 as below (A, B are sign bits for each s4 value).

AXXX BYYY

LSB: `<< 4` and `>> 4`, BBBB BYYY
MSB: `>> 4`, AAAA AXXX